src/Item_Disk.php

```diff
@@ -62,6 +62,43 @@ public static function getIcon()
         return 'far fa-hdd';
     }
 
+    /**
+     * Check if given object can have volume
+     *
+     * @since 10.1.0
+     *
+     * @param string|object $item  an object or a string
+     *
+     * @return boolean
+     *
+     **/
+    public static function canApplyOn($item): bool
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        if ($item instanceof CommonGLPI) {
+            $item = $item->getType();
+        }
+
+        return in_array($item, $CFG_GLPI['disk_types']);
+    }
+
+    /**
+     * Get all the types that can have volumes
+     *
+     * @since 10.1.0
+     *
+     * @return array itemtypes names
+     **/
+    public static function getItemtypesThatCanHave()
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        return $CFG_GLPI['disk_types'];
+    }
+
```

Comment:
I am not sure that these new methods will be used. Maybe it is not necessary to add them.